src/geom/color.rs

```diff
@@ -68,6 +70,27 @@ impl Color {
             Self::Cmyk(cmyk) => Self::Cmyk(cmyk.negate()),
         }
     }
+
+    /// Mixes multiple colors through weight.
+    pub fn mix(
+        colors: impl IntoIterator<Item = WeightedColor>,
+        space: ColorSpace,
+    ) -> StrResult<Color> {
+        let mut iter = colors.into_iter();
+        let WeightedColor(color, weight) =
+            iter.next().ok_or("expected at least one color")?;
+
+        let mut mixed = rgba_to_vec4(color.to_rgba(), space);
+        let mut total = weight;
+
+        for WeightedColor(color, weight) in iter {
+            let vec = rgba_to_vec4(color.to_rgba(), space);
+            total += weight;
+            mixed = lerp4(mixed, vec, weight / total);
```

Comment:
This is broken for `#color.mix((red, 0%), (yellow, 0%), green)` because we divide by zero.

Reply:
Is repeatedly lerping two values really the best way to interpolate with weights?

Reply:
Good catch. It's probably better to do something like

```math
\text{mixed} = \frac{\sum_i w_i \hat{v}_i}{\sum_i w_i}
```

```rs
let mut total = 0.0;
let mut acc = (0.0, 0.0, 0.0, 0.0);

for WeightedColor(color, weight) in colors.into_iter() {
    let v = rgba_to_vec4(color.to_rgba(), space);
    total += weight;
    acc.0 += weight * v.0;
    acc.1 += weight * v.1;
    acc.2 += weight * v.2;
    acc.3 += weight * v.3;
}

let mixed = (acc.0 / total, acc.1 / total, acc.2 / total, acc.3 / total);
Ok(vec4_to_rgba(mixed, space).into())
```

Reply:
Yeah that looks better. We still need to handle the case where all weights are zero / any result is NaN or inf though.